backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -485,6 +499,8 @@ def delete_dataset_bucket_key_policy(
                 principal_list.remove(f"{target_requester_arn}")
                 if len(principal_list) == 0:
                     statements.pop(DATAALL_ACCESS_POINT_KMS_DECRYPT_SID)
+                    if DATAALL_ACCESS_POINT_KMS_DECRYPT_SID not in statements.keys():
```

Comment:
do we need this if statement? If there are no principals in `principal_list` can we just remove `DATAALL_ACCESS_POINT_KMS_DECRYPT_SID`

We may need to keep `DATAALL_ACCESS_POINT_ENABLE_PIVOT_ROLE_PERMISSIONS_SID` Statement to avoid `MalformedPolicyDocument` error

Reply:
I am popping both Statements if principal list is 0 in the next revision. 
I dont think it will lead to MalformedPolicyDocument.. But we can test it out.

Reply:
testing the above while processing a bucket share worked with this PR, I did receive `MalformedPolicyDocumentException` when revoking a share 

Do we see any issue with checking if `DATAALL_ACCESS_POINT_ENABLE_PIVOT_ROLE_PERMISSIONS_SID` statement exists and if not just creating (i.e. never removing the permissions from the KMS Key Policy)

These are the required permissions we already implicitly give to the pivotRole in IAM with our conditional statements... I do not see an issue with doing as such but let me know your thoughts @anushka-singh @dlpzx

Reply:
If we do want to also keep the remove logic then can we provide a default value when we do `.pop()`, such as:

`statements.pop(DATAALL_BUCKET_ENABLE_PIVOT_ROLE_PERMISSIONS_SID, True)`

Otherwise we may run into `KeyErrors` for existing shares after they include the PR and they try to run a revoke

Reply:
In my opinion it makes sense to not remove it, because the pivot role access to the key is similar to a configuration requirement of a dataset that we apply in a lazy mode. 

If we keep it we avoid issues with simultaneous revoke/shares that create and delete it and with Malformedpolicy errors

Reply:
Okay I agree!
Good find @noah-paige 

In the remove functions, I will not pop `DATAALL_ACCESS_POINT_ENABLE_PIVOT_ROLE_PERMISSIONS_SID`